src/librbd/cache/SSDWriteLog.cc

```diff
@@ -0,0 +1,132 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "SSDWriteLog.h"
+#include "include/buffer.h"
+#include "include/Context.h"
+#include "include/ceph_assert.h"
+#include "common/deleter.h"
+#include "common/dout.h"
+#include "common/environment.h"
+#include "common/errno.h"
+#include "common/WorkQueue.h"
+#include "common/Timer.h"
+#include "common/perf_counters.h"
+#include "librbd/ImageCtx.h"
+#include "librbd/cache/rwl/ImageCacheState.h"
+#include "librbd/cache/rwl/LogEntry.h"
+#include <map>
+#include <vector>
+
+#undef dout_subsys
+#define dout_subsys ceph_subsys_rbd_rwl
+#undef dout_prefix
+#define dout_prefix *_dout << "librbd::cache::SSDWriteLog: " << this << " " \
+                           <<  __func__ << ": "
+
+namespace librbd {
+namespace cache {
+
+using namespace librbd::cache::rwl;
+
+template <typename I>
+SSDWriteLog<I>::SSDWriteLog(I &image_ctx, librbd::cache::rwl::ImageCacheState<I>* cache_state)
+  : ParentWriteLog<I>(image_ctx, cache_state)
+{
+}
+
+template <typename I>
+void SSDWriteLog<I>::update_pool_root(std::shared_ptr<WriteLogPoolRoot> root,
+                                      AioTransContext *aio) {
+  bufferlist bl;
+  super_block_t superblock;
+  superblock.root = *root;
+  superblock.checksum = 0;
+  encode(superblock, bl);
+  bl.append_zero(MIN_WRITE_ALLOC_SIZE - bl.length());
+  ceph_assert(bl.length() % MIN_WRITE_ALLOC_SIZE == 0);
+  bdev->aio_write(0, bl, &aio->ioc, false, WRITE_LIFE_NOT_SET);
```

Comment:
Where is `bdev` being initialized?

Reply:
It gets initialized in the `init` function which is not in this PR. Will push it with the next update.